### PR TITLE
Fix errors with rust v1.83

### PR DESCRIPTION
**Pull request by vigh-m**
*Description of changes:*
* Silence newly added linter errors and warnings
* Minor code changes based on linter warnings
* Update licenses to for `unicode-ident` crate

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: bottlerocket-settings-models/modeled-types/src/kubernetes.rs [hunk 1]
```diff
@@ -670,7 +670,7 @@ mod test_reserved_resources_key {
 }
 
 /// // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
-
+///
```
Comment (cbgbt) [inline bottlerocket-settings-models/modeled-types/src/kubernetes.rs]: I think we can remove the `///` from line 672.

File: bottlerocket-settings-plugin/src/settings.rs [hunk 2]
```diff
@@ -3,12 +3,11 @@ This crate defines the FFI specification for Bottlerocket settings plugins, as w
...
 
-// Avoid empty doc comment warning that originates from the StableAbi derive macro.
-#![allow(clippy::empty_docs)]
 // Avoid false positive improper ctypes warnings for abi_stable's PhantomData markers. We rely on
 // the StableAbi trait to catch any real problems.
-#![allow(improper_ctypes_definitions)]
-
+#![expect(improper_ctypes_definitions)]
+#![expect(non_local_definitions)]
+#![expect(clippy::needless_lifetimes)]
```
Comment (cbgbt) [inline bottlerocket-settings-plugin/src/settings.rs]: Do you mind elaborating that these also come from the `StableAbi` derive macro?

**Review by cbgbt — APPROVED** [review]

**Comment by vigh-m** [comment]
Updated based on comments.

File: deny.toml [hunk 3]
```diff
@@ -20,7 +20,7 @@ allow = [
 
 exceptions = [
     { name = "generational-arena", allow = ["MPL-2.0"] },
-    { name = "unicode-ident", allow = ["MIT", "Apache-2.0", "Unicode-DFS-2016"] },
+    { name = "unicode-ident", allow = ["MIT", "Apache-2.0", "Unicode-DFS-2016", "Unicode-3.0"] },
```
Comment (koooosh) [inline deny.toml]: I also have a PR which is failing CI due to this issue. In the CI build, I see this:
```
registry+https://github.com/rust-lang/crates.io-index#unicode-ident@1.0.14:4:36
  │
4 │ license = "(MIT OR Apache-2.0) AND Unicode-3.0"
```

Given this, can we remove the `"Unicode-DFS-2016"`?

Comment (vigh-m) [reply deny.toml]: Makes sense. Updated

**Comment by vigh-m** [comment]
Removed extra `"Unicode-DFS-2016"` license from `deny.toml`

**Review by koooosh — APPROVED** [review]

**Comment by koooosh** [comment]
Merging this to unblock my PR 😄

**Pull request merged**